testnet/stacks-node/src/nakamoto_node/relayer.rs

```diff
@@ -322,17 +467,18 @@ impl RelayerThread {
             "burn_height" => sn.block_height,
             "winning_txid" => %sn.winning_block_txid,
             "committed_parent" => %committed_index_hash,
+            "last_tenure_start_id" => %highest_tenure_start_block_header.index_block_hash(),
             "won_sortition?" => won_sortition,
         );
 
         if won_sortition {
             increment_stx_blocks_mined_counter();
         }
 
-        if sn.sortition {
+        let directive = if sn.sortition {
             if won_sortition {
                 MinerDirective::BeginTenure {
-                    parent_tenure_start: committed_index_hash,
+                    parent_tenure_start: highest_tenure_start_block_header.index_block_hash(),
```

Comment:
The miner must unambiguously begin mining from the tenure that they committed to. Not the highest tenure start block.

Reply:
In what world would it be the case that:

* `highest_tenure_start_block_header` is _not_ the tenure that the miner had committed to, and
* mining off of a fork that did not include `highest_tenure_start_block_header` is the correct thing to do?

Reply:
The miner simply cannot produce valid blocks if they begin their tenure building off of anything but the tenure they committed to. If you want to ensure that `committed_index_hash` is equal to `highest_tenure_start_block_header`, add a panicking assert. Any blocks assembled this way would be rejected during block processing (when the committed to hash is checked).

> In what world would it be the case that `highest_tenure_start_block_header` is not the tenure that the miner had committed to

Miner A wins tenure in bitcoin block 2.
< Time Passes >
Miner B submits a block commit, pointing at the tenure in bitcoin block 1 because Miner A still hasn't produced a block yet.
Miner A submits block proposal.
Signer set signs off on Miner A's block < -- > Bitcoin block 3 is produced (i.e., these events occur at nearly the same time)
Miner B sees Miner A's block, but cannot RBF their block commit anymore.
Miner B wins a tenure in bitcoin block 3.


In such a situation, Miner B must begin building blocks in their tenure off of bitcoin block 1's tenure. They don't have another option (other than just not mining).

Reply:
yup, see that now